dashboard/views.py

```diff
@@ -134,6 +137,90 @@ def get_course_info(request, course_id=0):
     return HttpResponse(json.dumps(resp, default=str))
 
 
+@permission_required('dashboard.update_course_info',
+                     fn=objectgetter(Course, 'course_id', 'canvas_id'), raise_exception=True)
+def update_course_info(request, course_id=0):
+    """
+
+    :param request: HTTP `PUT` req.; body should contain the JSON body…
+    :param course_id: Integer Canvas course ID number, typically six digits or less.
+    :return: HttpResponse containing `{"default": "success"}` or `{"default": "fail"}`
+    """
+    logger.info(update_course_info.__name__)
+
+    if (request.method != 'PUT'):
+        return HttpResponseBadRequest(JsonResponse({'error': 'Invalid request method.'}))
+
+    course_id = canvas_id_to_incremented_id(course_id)
+    current_user = request.user.get_username()
+
+    bad_json_response = HttpResponseBadRequest(JsonResponse({'error': 'Request JSON malformed.'}))
+
+    try:
+        request_data: dict = json.loads(request.body.decode('utf-8'))
+    except JSONDecodeError:
+        return bad_json_response
+
+    schema = {'$schema': 'http://json-schema.org/draft-07/schema',
+              'type': 'object',
+              'additionalProperties': False,
+              'properties': {
+                  'ap': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'}}},
+                  'apv1': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'}}},
+                  'gd': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'},
+                                     'show_grade_counts': {'type': 'boolean'}}},
+                  'ra': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'}}}},
+              'minProperties': 1}
```

Comment:
I liked the way `jsonschema` helped out here.  I would liked to have skipped the repetition of `'additionalProperties': False` everywhere, though.  I asked around and there are supposedly ways to define it once near the beginning of the schema and have it apply to all objects, but I couldn't get it to work.  I think `jsonschema` doesn't support the version of the schema that allows that, but also, I couldn't get it to work with other online JSON Schema tools.